### PR TITLE
[docs] Introduce tutorial on writing reftests

### PR DESCRIPTION
**Pull request by jugglinmike**
Tutorials are a proven way of helping potential contributors orient
themselves quickly. By providing concrete instructions that are focused
on one specific task, projects can give readers a sense of the
contribution experience and offer a starting point for learning any
additional details that are relevant to their particular goals.

Replace existing content which demonstrates the contents of a reftest
with a complete narrative describing how and why such a test is
authored.

---

This was inspired by conversations with @fantasai, @frivoal, @lmccart, @ericholscher, @jihyerish, and @chrisdavidmills and [the experience of the Django project maintainers](https://jacobian.org/2009/nov/10/what-to-write/).

Since this is more objective than any of the content we've considered so far (and since it's just a first draft), I especially welcome high-level design feedback. For example: is the `<bdo>` element a good motivating feature? Does this try to cover too much? Does it not cover enough?

I'm planning to write a similar tutorial for [testharness.js tests](https://web-platform-tests.org/writing-tests/testharness.html) and potentially still more tutorials after that. As that content comes together, I'd like to reference it from [the documentation's landing page](https://web-platform-tests.org/), but for now, I've limited integration to a couple places in the "writing tests" section.

### CODE REVIEW

File: docs/writing-tests/reftest-tutorial.md
Comment (sideshowbarker) [inline docs/writing-tests/reftest-tutorial.md]:
```suggestion
  here, it might seem a liitle superfluous. It's much more helpful for more-
```

File: docs/writing-tests/reftest-tutorial.md
Comment (sideshowbarker) [inline docs/writing-tests/reftest-tutorial.md]:
```suggestion
WPT's reftests are great for testing web-platform features that have some
```

File: docs/writing-tests/reftest-tutorial.md
Comment (sideshowbarker) [inline docs/writing-tests/reftest-tutorial.md]:
```suggestion
done. However, it's time-consuming for a human to run tests, so we should
```

File: docs/writing-tests/reftest-tutorial.md
Comment (sideshowbarker) [inline docs/writing-tests/reftest-tutorial.md]:
```suggestion
don't), using `git diff`:
```

File: docs/writing-tests/reftests.md
Comment (sideshowbarker) [inline docs/writing-tests/reftests.md]:
```suggestion
on writing a reftest](reftest-tutorial) offers a more-limited but
```

**Review by sideshowbarker — APPROVED** [review]
Beautiful

File: docs/writing-tests/reftest-tutorial.md
Comment (jugglinmike) [reply docs/writing-tests/reftest-tutorial.md]: This matches the way I would enunciate the sentence, but I don't see how it's grammatically necessary. If the parenthetical weren't there, then the comma would feel more out of place.

Though scrutinizing the sentence along those lines makes me think that the word "by" is necessary here.

File: docs/writing-tests/reftests.md
Comment (jugglinmike) [reply docs/writing-tests/reftests.md]: I'm clearly not accustomed to hyphenating compound adjectives, but it seems to me that this hyphen is superfluous due to the word "a". For instance, the phrases "offers more limited choices" and "offers more-limited choices" mean
two different things, but the phrases "offers a more limited set of choices" and "offers a more-limited set of choices" do not.

File: docs/writing-tests/reftest-tutorial.md
Comment (sideshowbarker) [reply docs/writing-tests/reftest-tutorial.md]: Actually to me, if it had *“We can make sure the commit has everything we want to submit using `git diff`”*, it’s even more ambiguous: Does it mean there is some set of things you want to use `git diff` to submit? Or does it instead mean, *Using `git diff`, we can make sure the commit has everything we want to submit.*? Of course it means the latter, but without the comma — or without moving the *Using `git diff`* to the beginning of the sentence instead of the end — it’s ambiguous. But I don’t feel strongly about it needing to be changed. I defer to your judgment on  whether you think it’s fine as-is. Like a lot of copy edits, it’s just a judgment call.

File: docs/writing-tests/reftests.md
Comment (sideshowbarker) [reply docs/writing-tests/reftests.md]: Hyphenating compound adjectives is basically never strictly necessary. I agree it would be more warranted if the word "a" weren’t there. But it’s never harmful to hyphenate a compound adjective. It certainly doesn’t ever make the meaning less ambiguous. It just helps for the case where somebody might otherwise trip over when reading. But I don’t feel strongly about it. Judgment call (yours — since you’re the one who did the work to write the actual content).

File: docs/writing-tests/reftest-tutorial.md
Comment (sideshowbarker) [inline docs/writing-tests/reftest-tutorial.md]:
```suggestion
  here, it might seem a little superfluous. It's much more helpful for
```

**Review by sideshowbarker — APPROVED** [review]

File: docs/writing-tests/reftest-tutorial.md
Comment (jugglinmike) [reply docs/writing-tests/reftest-tutorial.md]: Now you've gone too far. Spelling mistakes give writing personality!

Comment (sideshowbarker) [reply docs/writing-tests/reftest-tutorial.md]: haha — I wasn’t sure if you were actually misspelled *leetle* but I took a shot in the dark and guessed it might be *little* instead

**Comment by jugglinmike** [comment]
@sideshowbarker I've made a few changes based on my experience writing [another tutorial](https://github.com/web-platform-tests/wpt/pull/17894):

- Avoid instructing the reader to actually create a pull request
- Explain the design goal of conciseness over completeness as an HTML comment for future maintainers (this might go better in a dedicated document when/if both tutorials are accepted)
- Add some suggestions for self-study

File: docs/writing-tests/reftest-tutorial.md
Comment (sideshowbarker) [inline docs/writing-tests/reftest-tutorial.md]:
```suggestion
the reader with features. Because typical WPT usage patterns change over time,
```

File: docs/writing-tests/reftest-tutorial.md
Comment (sideshowbarker) [inline docs/writing-tests/reftest-tutorial.md]:
```suggestion
this should be updated periodically; but please weigh extensions against the demotivating effect that a
```

File: docs/writing-tests/reftest-tutorial.md
Comment (gsnedders) [inline docs/writing-tests/reftest-tutorial.md]: Let's drop this link; there's no real need for it.

Comment (jugglinmike) [reply docs/writing-tests/reftest-tutorial.md]: Non-essential metadata is gratuitous for this guide. Good call

**Review by gsnedders — APPROVED** [review]

**Comment by jugglinmike** [comment]
Thanks, @sideshowbarker! Thanks, @gsnedders! I'll do the honors myself

**Pull request merged**
